runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -251,14 +249,39 @@ ILogData read(long address) {
                 // is much cheaper than the cost of a NoRollBackException, therefore
                 // this trade-off is reasonable
                 final ILogData loadedVal = fetch(address);
-                data = readCache.asMap().computeIfAbsent(address, (k) -> loadedVal);
-                return data;
-            } else {
-                return data;
+                return readCache.asMap().computeIfAbsent(address, (k) -> loadedVal);
             }
-        } else {
-            return fetch(address);
+
+            return data;
         }
+
+        return fetch(address);
+    }
+
+    /**
+     * This method reads a batch of addresses if 'nextRead' is not found in the cache.
+     * In the case of a cache miss, it piggybacks on the read for nextRead.
+     *
+     * If 'nextRead' is present in the cache, it directly returns this data.
+     *
+     * @param nextRead current address of interest
+     * @param addresses batch of addresses to read (bring into the cache) in case there is a cache miss (includes
+     *                  nextRead)
+     * @return data for current 'address' of interest.
+     */
+    public @Nonnull ILogData predictiveReadRange(Long nextRead, List<Long> addresses) {
```

Comment:
Is it better to change this method to private? It seems that this method should not be exposed.